### ISSUE TITLE
Should we manually apply weight_decay to optimizers besides AdamW?

### ISSUE THREAD

**Issue by jarednielsen** [tensorflow/addons#1902]
Reading the [AdamW documentation](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/AdamW), I noticed `Note: when applying a decay to the learning rate, be sure to manually apply the decay to the weight_decay as well.`. 

Does this also apply to other optimizers (like [LAMB](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/LAMB)), or is this a quirk specific to AdamW?

**Comment by jarednielsen** [comment]
As detailed in the [AdamW paper](https://arxiv.org/pdf/1711.05101.pdf), it only applies to decoupled weight regularization. In traditional L2 regularization, the regularization occurs in the gradient and thus the learning rate decay applies to it. Decoupled weight regularization moves the weight regularizer outside of the gradient and thus it needs its own schedule. Sorry for raising an issue before doing sufficient research.
